app/controllers/checkout_controller.rb

```diff
@@ -261,6 +253,13 @@ def update_failed_response
     end
   end
 
+  def rescue_from_spree_gateway_error(error)
+    flash[:error] = t(:spree_gateway_error_flash_for_checkout, error: error.message)
+    # This can also happen during the edit action
+    #   but the actions and response needed are exactly the same as when the update action fails
+    update_failed(error)
+  end
```

Comment:
the key difference here is the call to checkout_failed in update_failed that executes PostCheckoutActions that will reset the order state to cart :+1: